apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/service/blockchain/BlockchainImpl.java

```diff
@@ -121,10 +134,12 @@ public void setLastBlock(Block block) {
     @PostConstruct
     @Override
     public void update() {
-        this.lastBlock.set(findLastBlock());
-        this.shardInitialBlock.set(findFirstBlock());
-        ((ShardManagement) this.databaseManager).initFullShards(
-            shardDao.getAllCompletedShards().stream().map(Shard::getShardId).collect(Collectors.toList()));
+        if (isPostConstruct) { // for unit tests !
```

Comment:
Let's discus about this case.  Why do we need it if we use @PostConstruct annotation.

Reply:
As I think junit5 implementation has changed and now @PostContruct is get called by weld container inside unit test.
That gives a problem for instantiation of BlockchainImpl component when unit test is instantiated.

if you change code in **BlockchainTest**
replacing 
public WeldInitiator weld = WeldInitiator.from(.....**BlockchainProducerUnitTests**.class,
by class
**BlockchainProducer**.class,
you'll see the main problem for many unit tests.